src/py42/clients/_settings_managers.py

```diff
@@ -0,0 +1,520 @@
+from py42._internal.compat import str
+from py42._internal.compat import UserList
+from py42._internal.compat import UserDict
+from py42.exceptions import Py42Error
+
+
+def set_val(d, keys, value):
+    """Helper for setting nested values from a dict based on a list of keys."""
+    d = get_val(d, keys[:-1])
+    d[keys[-1]] = value
+
+
+def get_val(d, keys):
+    """Helper for getting nested values from a dict based on a list of keys."""
+    for key in keys:
+        d = d[key]
+    return d
+
+
+def bool_to_str(value):
+    if isinstance(value, bool):
+        return "true" if value else "false"
+    elif value in ("true", "false"):
+        return value
+    else:
+        raise ValueError("Value must be True or False")
+
+
+def str_to_bool(value):
```

Comment:
Perhaps these should be internal prefixed? Else I am tempted to re-comment that you should call `lower()`

Reply:
Otherwise, these might be more generally useful and could be add to `util`